src/components/InjectedComponents/AdditionalPostBox.tsx

```diff
@@ -113,7 +117,10 @@ export function AdditionalPostBox(props: Partial<Props>) {
         [identity],
     )
 
-    if (identity.length === 0) {
+    const [showWelcome, setShowWelcome] = useState(false)
+    useAsync(getActivatedUI().shouldDisplayWelcome, []).then(x => setShowWelcome(x))
+
+    if (showWelcome) {
```

Comment:
what about check `identity.length === 0` and `showWelcome` at the same time?

Reply:
Going to check the currently user and ```shouldDisplayWelcome``` in the future. Is a todo in stage 4 (https://github.com/DimensionDev/Maskbook/pull/228)